rotkehlchen/rotkehlchen.py

```diff
@@ -903,3 +915,38 @@ def create_oracle_cache(
             return  # only for cryptocompare for now
 
         self.cryptocompare.create_cache(from_asset, to_asset, purge_old)
+
+    def query_missing_prices(
+        self,
+        filter_query: HistoryEventFilterQuery,
+    ) -> None:
+        """Queries missing prices for HistoryBaseEntry in database based
+        on the query filter used"""
+        # Use a deepcopy to avoid mutations in the filter query if it is used later
+        new_filter_query = copy.deepcopy(filter_query)
+        new_filter_query.filters.append(DBStringFilter(and_op=True, column='usd_value', value='0'))
+        entries_missing_prices = self.data.db.get_rows_missing_prices_in_base_entries(
+            filter_query=new_filter_query,
+        )
+        inquirer = PriceHistorian()
+        updates = []
+        for identifier, amount, asset, timestamp in entries_missing_prices:
+            price = inquirer.query_historical_price(
+                from_asset=asset,
+                to_asset=A_USD,
+                timestamp=timestamp,
+            )
+            usd_value = amount * price
+            updates.append((str(usd_value), identifier))
+
+        self.data.db.update_base_entries_prices(updates)
+
+    def periodically_query_history_events_prices(self, sleep_time_secs: int) -> None:
+        """
+        Query periodically the history_events table to find entries missing a proper
+        usd_value column value.
+        """
+        query = HistoryEventFilterQuery.make()
+        while True:
+            self.query_missing_prices(filter_query=query)
+            gevent.sleep(sleep_time_secs)
```

Comment:
And no absolutely not like that.

This will be a task that runs forever and ever if even 1 price for 1 asset at a single timestamp can not be found. Checks tasks/manager.py